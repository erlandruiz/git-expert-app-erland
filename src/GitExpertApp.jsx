import { useState } from "react";
import { AddCategory , GifGrid} from "./components";


export const GitExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);

  const onAddCategory = (newcategory) => {
    if (categories.includes(newcategory)) return;
    // console.log(newcategory);
    setCategories([newcategory, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>

      {/* Input */}

      <AddCategory onNewCategory={(value) => onAddCategory(value)} />
      {/* <AddCategory setCategories = {setCategories}/> */}

      {/* Listado de Gif */}
      {/* <button onClick={onAddCategory}>Agregar</button> */}

      {/* <ol>
        {categories.map((category) => {
          return <li key={category}>{category}</li>;
        })}
      </ol> */}

     
      {
        categories.map((category) => (
          <GifGrid 
            key={category} 
            category={category} 
          />
        ))
        
      }
      
    </>
  );
};
