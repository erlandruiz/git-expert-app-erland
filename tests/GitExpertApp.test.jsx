import { render, screen } from "@testing-library/react"
import { GitExpertApp } from "../src/GitExpertApp"


describe('Primeras pruebas en componente <GitExpertApp/>', () => { 
    test('deberia hacer match con el snapsshot', () => { 
       const{container} = render(<GitExpertApp/>);
       screen.debug();
        expect(container).toMatchSnapshot();
        
    });

    
 });