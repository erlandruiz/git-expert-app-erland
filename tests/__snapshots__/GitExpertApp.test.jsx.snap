// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primeras pruebas en componente <GitExpertApp/> deberia hacer match con el snapsshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando..
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
